### DISCUSSION TITLE
How to initialize Shiny in Uno Application?

### DISCUSSION THREAD

**Discussion by pdecostervgls** [shinyorg/shiny#825] [Q&A]
My app is based on Uno Platform, therefore the App.xaml.cs file is shared across all heads. If I add the `[assembly: Shiny.ShinyApplication...]` directive it says its only available on UWP.

I also tried to call the `Shiny.ShinyHost.Init(IPlatform platform, IShinyStartup? startup)` method, but I dont know how to get the value I have to pass as `IPlatform`.

**Comment by aritchie** [comment]
There is no source generation that happens on UWP.  I'm afraid I don't know enough about Uno at this time in order to be able help.  

I'll add it as a todo over in the sample repo.  You can also create these types of requests going forward here: https://github.com/shinyorg/samples/issues/new/choose

**Reply by pdecostervgls** [reply]
That's too bad, because with our internally forked version, the `Shiny.UwpShinyHost.Init(` used to work.

**Reply by aritchie** [reply]
There is a bug with 2.4 missing UWP targets.  2.4.1 is building for production right now.  Ensure your UWP target is set to 17764 (or whatever it is)
